docs/user/interactive_stopping_and_saving.rst

```diff
@@ -0,0 +1,119 @@
+(both available in JupyterLab and RStudio), and behind the scenes both are using ``git``
+staging (``add``), ``commit``, and ``push``. You can type these commands directly
+into the available terminal interface of your interactive environment, or click
+some buttons via the git plugins.
+
+When you push your changes back to RenkuLab, the GitLab CI/CD is triggered to build
+a new image out of the ``Dockerfile``, which will be available the next time you
+start a new environment.
+
+Saving via Terminal
+~~~~~~~~~~~~~~~~~~~
+
+In the Terminal interface inside the interactive environment, call the following
+three ``git`` commands to stage, commit, and push your changes:
+
+1. ``git add *``
+2. ``git commit -m "my short but descriptive message of the changes I made"``
+3. ``git push``
+
+If you are new to git, these resources might be useful:
+
+* [git documentation](https://git-scm.com/doc)
+* [A great interactive cheat-sheet](http://ndpsoftware.com/git-cheatsheet.html)
+* [An intro to git](https://rogerdudler.github.io/git-guide/)
+
+Saving via Git Plugin
+~~~~~~~~~~~~~~~~~~~~~
+
+Find the git plugin interface (Jupyterlab: branched-dots icon on lefthand vertical
+menu; RStudio: top right box). Add the changed files you want to save to staging,
+write a message to commit the changes, and don't forget to hit the icon or button
+to push those changes.
+
+.. _autosave:
+
+Autosave in JupyterLab
+----------------------
+
+When you stop a JupyterLab notebook, an automatic check looks for any work
+that has not been pushed to the remote repository, including untracked and
+modified files. If something is found, a new "autosave" branch is created
+and pushed to GitLab.
+
+.. note::
+
+  You cannot start a notebook server from these branches but you can
+  still see them listed in GitLab. You can easily identify them because
+  their name starts with `renku/autosave/`
+
+
+Restore unsaved work
+~~~~~~~~~~~~~~~~~~~~
+
+The next time you start a new JupyterLab server from the same branch/commit
+combination, you will be notified and the autosaved data will be automatically
+loaded in your session. Please note that nothing will be pushed automatically
+to the origin, therefore you won't see any changes in your project's files
+in Renku, nor will the :ref:`provenance <provenance>` be updated.
+
+Local branches you created but never pushed to the origin will not be saved.
+
+
+Discard autosaved content
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+If you don't need the autosaved content, you can easily discard it using the
+following commands:
+
+::
+
+    $ git reset --hard origin/<branch-name> # remove commits and modified files
+    $ git clean -f -d # remove untracked files
+
+If you are working on the ``master`` branch, the command looks like this:
+
+::
+
+    $ git reset --hard origin/master && git clean -f -d
+
+
+Push changes regularly
+~~~~~~~~~~~~~~~~~~~~~~
+
+This feature is intended to prevent losing your work but you should not rely
```

Comment:
Suggestion:
```
The autosave feature is intended to prevent loss of work, but it is not
a replacement for `git`. By making commits in `git`, you track your
changes, and pushing back to RenkuLab ensures that they are 
visible to others. And although we are always working to improve
robustness, in some situations, autosave can fail. So the most
secure way to keep your work is to commit and push to origin.
```